src/app/services/JoinerService.ts

```diff
@@ -34,18 +25,25 @@ export class JoinerService {
         display(JoinerService.VERBOSE, 'JoinerService.createInitialJoiner(' + creatorName + ', ' + joinerId + ')');
 
         const newJoiner: IJoiner = {
-            ...JoinerService.EMPTY_JOINER,
+            candidates: [],
+            chosenPlayer: null,
+            firstPlayer: FirstPlayer.RANDOM.value,
+            partType: PartType.STANDARD.value,
+            partStatus: PartStatus.PART_CREATED.value,
+            maximalMoveDuration: PartType.NORMAL_MOVE_DURATION,
+            totalPartDuration: PartType.NORMAL_PART_DURATION,
             creator: creatorName,
         };
         return this.set(joinerId, newJoiner);
     }
     public async joinGame(partId: string, userName: string): Promise<boolean> {
         display(JoinerService.VERBOSE, 'JoinerService.joinGame(' + partId + ', ' + userName + ')');
 
-        const joiner: IJoiner = await this.joinerDao.read(partId);
-        if (joiner == null) {
+        const joinerOpt: MGPOptional<IJoiner> = await this.joinerDao.tryToRead(partId);
```

Comment:
let's avoid "const fooType: Type" declaration :p (joiner is only used once, it's deletable)

Reply:
t'as une proposition pour quand on a un MGPOptional à checker puis à .getter pour l'utiliser plus d'une fois ? (ce pattern arrive plusieurs fois dans le code)

Reply:
Ouais pour ceux y a pas le choix, et pour ceux là p-e exceptionnellement accepter Opt plutôt que Optional, mais privilégier les alternatives ou on doit pas trop faire ça :p mais là on sais faire sans !

Reply:
il est utilisé deux fois ici ! Mais ajouté des .get() aux deux utilisations du coup.